docs/users/cross-compilation.md

```diff
@@ -0,0 +1,144 @@
+```
+
+`common.conf`
+```scala
+rules = [
+  DisableSyntax,
+  OrganizeImports
+]
+
+DisableSyntax {
+  noFinalize = true
+}
+```
+
+`scala2.conf`
+```scala
+include "common.conf"
+
+rules += RemoveUnused
+
+// Scala 2 only compilers flags or rule settings
+RemoveUnused {
+  imports = true
+}
+```
+
+`scala3.conf`
+```scala
+include "common.conf"
+
+rules += LeakingImplicitClassVal
+
+// Scala 3 specific tweaks go here
+OrganizeImports {
+  groupedImports = Keep
+}
+```
+
+### Multiple include files
+
+You may split out even more granular snippets (for example `linting.conf`,
+`rewrites.conf`) and include them from both `scala2.conf` and `scala3.conf`. The
+HOCON syntax supports nested includes, so feel free to create the hierarchy that
+matches your team conventions.
+
+## Selecting the right config in sbt
+
+Point `scalafixConfig` at the version-specific file, typically inside a helper
+setting applied to all cross-built projects:
+
+```scala
+import scalafix.sbt.ScalafixPlugin.autoImport._
+
+lazy val commonSettings = Seq(
+  scalafixConfig := {
+    val base = (ThisBuild / baseDirectory).value / "scalafix"
+    val file =
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((3, _)) => base / "scala3.conf"
+        case _            => base / "scala2.conf"
+      }
+    Some(file)
+  }
+)
+
+lazy val core = project
+  .settings(commonSettings)
+  .settings(
+    scalaVersion := "3.3.3",
+    crossScalaVersions := Seq("2.13.14", "3.3.3")
+  )
+```
+
+For builds that already differentiate per configuration (`Compile`, `Test`,
+`IntegrationTest`), you can set `Compile / scalafixConfig` and
+`Test / scalafixConfig` separately if the inputs diverge.
+
+### Command-line usage
+
+When invoking the CLI directly, pass the desired config with `--config`:
+
+```
+scalafix --config scalafix/scala2.conf --rules RemoveUnused
+scalafix --config scalafix/scala3.conf --rules LeakingImplicitClassVal
+```
+
+Your CI job can loop over each target Scala version, selecting the matching
+config before running `scalafix --check`.
+
+## Recommendations and current limitations
+
+* Keep rules that truly work on both versions inside `common.conf`.
+* Isolate risky scalac options (`-Wunused`, `-Ywarn-unused-import`, `-P:semanticdb`)
+  inside each version-specific file or sbt setting.
+* Document in the repository README which rules run on which Scala version to
+  reduce confusion for new contributors.
+
+### Looking ahead
+
+Issue [#1747](https://github.com/scalacenter/scalafix/issues/1747) tracks better
+ergonomics for cross compilation. Potential improvements include:
+
+* Conditional configuration blocks directly inside `.scalafix.conf` (for example
+  `if scalaVersion.startsWith("3.")`).
+* First-class support for including multiple files via CLI flags.
+* Allowing rule selection based on the detected input Scala dialect.
+
+Until those land, the include-based layout above is the recommended, battle-tested
+approach.
+
```

Comment:
I don't think we should advertize potential improvements in the docs, as it's unclear if there is a need for it. As a matter of fact, I was hoping to close https://github.com/scalacenter/scalafix/issues/1747 with this doc update.